cmd/virt-chroot/tap-device-maker.go

```diff
@@ -65,11 +75,21 @@ func NewCreateTapCommand() *cobra.Command {
 				return fmt.Errorf("could not parse tap device group: %v", err)
 			}
 
-			if err := createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu)); err != nil {
-				return fmt.Errorf("failed to create tap device named %s. Reason: %v", tapName, err)
-			}
-
-			return nil
+			return createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu))
 		},
 	}
 }
+
+func retry(retryAttempts uint, f func() error) (uint, error) {
+	var errorsString []string
+	for attemptID := uint(0); attemptID < retryAttempts; attemptID++ {
+		if err := f(); err != nil {
+			errorsString = append(errorsString, fmt.Sprintf("[%d]: %v", attemptID, err))
+			time.Sleep(time.Second)
+		} else {
+			return attemptID, nil
```

Comment:
If the `errorsString` is not empty, I would log it as a warning.

Reply:
Done

Reply:
I would extract `strings.Join(errorsString, "\n")` to a variable.